### PR TITLE
Memoize renderText function in Message components

### PR DESCRIPTION
**Pull request by vicnicius**
## Description of the pull request

Memoize renderText function in Message components.

### CODE REVIEW

File: src/components/Message/MessageText.js [hunk 1]
```diff
@@ -40,6 +40,16 @@ const MessageTextComponent = (props) => {
   const hasReactions = messageHasReactions(message);
   const hasAttachment = messageHasAttachments(message);
   const handleReaction = useReactionHandler(message);
+  const messageText = useMemo(() => {
+    if (!message?.text) {
+      return null;
+    }
```
Comment (mahboubii) [inline src/components/Message/MessageText.js:46]: the null check exists on the renderText fn

File: src/components/Message/MessageText.js [hunk 2]
```diff
@@ -106,9 +116,14 @@ const MessageTextComponent = (props) => {
...
       <MessageOptions
-        {...props}
-        {...customOptionProps}
         onReactionListClick={onReactionListClick}
+        initialMessage={props.initialMessage}
+        message={message}
+        messageWrapperRef={props.messageWrapperRef}
+        threadList={props.threadList}
+        getMessageActions={props.getMessageActions}
+        handleFlag={props.handleFlag}
```
Comment (mahboubii) [inline src/components/Message/MessageText.js:125]: 🥇

File: src/utils.js [hunk 3]
```diff
@@ -166,13 +166,12 @@ export const truncate = (input, length, end = '...') => {
   return input;
 };
 
-export const renderText = (message) => {
+export const renderText = ({ text, mentioned_users }) => {
```
Comment (mahboubii) [inline src/utils.js:169]:
```suggestion
export const renderText = (text, mentioned_users) => {
```

File: src/components/Message/MessageText.js (hunk 2 above)
Comment (vicnicius) [reply src/components/Message/MessageText.js]: hehe. It broke some types. We need some adjustments in other places so I'll tackle this on a different PR. I reverted here: https://github.com/GetStream/stream-chat-react/pull/452/commits/9aa529462d7187c695586248842d5811880441ca.

File: src/utils.js (hunk 3 above)
Comment (vicnicius) [reply src/utils.js]: 👌  Improved here: https://github.com/GetStream/stream-chat-react/pull/452/commits/4cc8993e3804b8fee46d3cb95fd6816a518ae044

File: src/components/Message/MessageText.js (hunk 1 above)
Comment (vicnicius) [reply src/components/Message/MessageText.js]: 👍  Improved here: https://github.com/GetStream/stream-chat-react/pull/452/commits/4cc8993e3804b8fee46d3cb95fd6816a518ae044

File: src/components/Message/MessageText.js [hunk 4]
```diff
@@ -40,6 +40,11 @@ const MessageTextComponent = (props) => {
   const hasReactions = messageHasReactions(message);
   const hasAttachment = messageHasAttachments(message);
   const handleReaction = useReactionHandler(message);
+  const messageText = useMemo(
+    () => renderText(message?.text, message?.mentioned_users),
+    [message?.text, message?.mentioned_users],
+  );
+  console.log();
```
Comment (mahboubii) [inline src/components/Message/MessageText.js:47]:
```suggestion
```

**Pull request merged**